p2p/upgrade_test.go

```diff
@@ -46,3 +50,32 @@ func TestIsBootnode(t *testing.T) {
 		})
 	}
 }
+
+func TestConnectionsNotifier(t *testing.T) {
+	const n = 3
+	mesh, err := mocknet.FullMeshLinked(n)
+	require.NoError(t, err)
+	counter := [n]atomic.Uint32{}
+	// we count events - not peers
+	reporter := func(i int) func() {
+		return func() {
+			counter[i].Add(1)
+		}
+	}
+
+	for i, host := range mesh.Hosts() {
+		_, err := Upgrade(host, types.Hash20{}, WithNodeReporter(reporter(i)))
+		require.NoError(t, err)
+	}
```

Comment:
NIT - is the extra factory needed?
```suggestion
	for i, host := range mesh.Hosts() {
	       i := i
		_, err := Upgrade(host, types.Hash20{}, WithNodeReporter( func() { counter[i].Add(1) }))
		require.NoError(t, err)
	}
```

Reply:
thanks, makes sense